File: puremvc/puremvc/MVC/Patterns/Proxy/Proxy.cs
using MVC.Interfaces;
using MVC.Patterns.Observer;

namespace MVC.Patterns.Porxy
{
    public class Proxy: Notifier, IProxy
    {
        public const string NAME = "Proxy";

        public Proxy(string proxyName, object data = null)
        {
            ProxyName = proxyName ?? Name;
            if(data != null) Data = data;
        }

        public virtual void OnRegister()
        {
        }

        public string ProxyName{get; protected set;}

        public object Data {get; set;}
    }
}

File: framework/project/base/XScripts/Frameworks/MVC/Patterns/Proxy/Proxy.lua
local Proxy = Class("MVC.Proxy", MVC.Notifier)
Proxy.NAME = "Proxy"
function Proxy:Ctor(proxyName, data)
    self.proxyName = proxyName or Proxy.NAME
    if data ~= nil then
        self:SetData(data)
    end
end
function Proxy:GetProxyName()
    return self.proxyName
end
function Proxy:SetData(data)
    self.data = data
end
function Proxy:GetData()
    return self.data
end
function Proxy:OnRegister()
end
function Proxy:OnRemove()
end

return Proxy